tests/MongoDB.Driver.Tests/AuthenticationTests.cs

```diff
@@ -294,27 +295,45 @@ public void Authentication_succeeds_when_user_has_multiple_credentials_and_mecha
             var settings = client.Settings.Clone();
             settings.Credential = MongoCredential.FromComponents(
                 mechanism: "SCRAM-SHA-256", source: null, username: uniqueUnicodeUserName, password: unicodePassword);
-            
+
             AssertAuthenticationSucceeds(settings);
         }
+
 #endif
-        
+        [SkippableFact]
+        public void Authentication_succeeds_with_MONGODB_X509_mechanism()
+        {
+            RequireServer.Check().Tls(required: true);
+            RequireEnvironment.Check().EnvironmentVariable("MONGO_X509_CLIENT_P12_PATH", isDefined: true);
+            RequireEnvironment.Check().EnvironmentVariable("MONGO_X509_CLIENT_P12_PASSWORD", isDefined: true);
+            var pathToClientCertificate = Environment.GetEnvironmentVariable("MONGO_X509_CLIENT_P12_PATH");
+            var password = Environment.GetEnvironmentVariable("MONGO_X509_CLIENT_P12_PASSWORD");
+            var clientCertificate = new X509Certificate2(pathToClientCertificate, password);
+            var settings = DriverTestConfiguration.GetClientSettings().Clone();
+            // if a username is null, the distinguished name from the certificate will be used
+            settings.Credential = MongoCredential.CreateMongoX509Credential(username: null);
+            settings.SslSettings = settings.SslSettings.Clone();
```

Comment:
Cloning the SSL settings ensures we pick up the necessary values for `DisableCertificateRevocationChecking` that we set in `run-tests.sh`. We could also in just forcibly disable certificate revocation checking because for now, the certs used for non-ocsp testing don't have any revocation checking mechanisms defined.